### DISCUSSION TITLE
pybind11  "error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?"

### DISCUSSION THREAD

**Discussion by yentingl1** [pybind/pybind11#5232] [General]
I was trying to use cppimport to compile and import several cpp files. The first cpp file compiled but the second one gave an error about copy_n not being a member of 'std'. Any idea how to fix this?

##################################################################
In file included from /home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/pybind11/include/pybind11/eigen/../pybind11.h:19,
                 from /home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/pybind11/include/pybind11/eigen/../numpy.h:12,
                 from /home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/pybind11/include/pybind11/eigen/matrix.h:12,
                 from /home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/pybind11/include/pybind11/eigen.h:12,
                 from /home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/rpxdock/cluster/.rendered.cookie_cutter.cpp:6:
/home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/pybind11/include/pybind11/eigen/../typing.h: In constructor ‘constexpr pybind11::typing::StringLiteral<N>::StringLiteral(const char (&)[N])’:
/home/fedora_1/venvs/Rosetta_related/lib/python3.12/site-packages/pybind11/include/pybind11/eigen/../typing.h:104:58: error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?
  104 |     constexpr StringLiteral(const char (&str)[N]) { std::copy_n(str, N, name); }
      |                                                          ^~~~~~
      |                                                          copy
error: command '/usr/bin/gcc' failed with exit code 1
##################################################################

**Comment by SHiftLin** [comment]
I have the same error. I think it is probably because the header <algorithm> is not included

**Comment by yshi02** [comment]
There's another issue about this, see https://github.com/pybind/pybind11/issues/5206

There's no newer release of pybind11 yet, so the best workaround is to manually add `#include <algorithm>` to pybind11's `typing.h`.
